docs/dev.md

```diff
@@ -64,6 +70,71 @@ dep ensure
 /go/bin/deepcopy-gen --output-base=/go/src --input-dirs=github.com/kubeflow/experimental-kvc/pkg/apis/cr/v1/...
 go build -gcflags "-N -l" github.com/kubeflow/experimental-kvc
 ```
+### Developing on a workstation
+This is the preferred method for developers who don't want to run `docker` locally and or don't mind setting up
+`Go` development environment on their workstation.
+
+Prior to using this method please ensure you have `go 1.9.2` or better development environment setup and running.
+
+For developing on `Linux` add the following lines to your `.bashrc`
+```bash
+export GOROOT="/usr/lib/go-1.9"
+export GOPATH="$HOME/go"
+export PATH="$PATH:$GOROOT/bin:$GOPATH/bin"
+```
+
+and for `MacOSX` add these lines to your `.bashrc`:
+ ```bash
+export GOROOT="/usr/local/opt/go/libexec"
+export GOPATH=$HOME/go
+export PATH=$PATH:$GOROOT/bin:$GOPATH/bin
+```
+in both cases, install `mercurial` either using `yum` or `apt-get` or `brew`.
+This is because some dependencies live in `bitbucket` repositories.
+
+Then run these commands:
+```bash
+mkdir -p $GOPATH/src/github.com/kubeflow
```

Comment:
I don't think we should get specific here. It's up to the developer to make sure how they want to make the kvc repo available in the GOPATH. So can we just say.. make sure kvc is in `$GOPATH/src/github.com/kubeflow/experimental-kvc`?